druid-shell/src/platform/x11/window.rs

```diff
@@ -124,36 +142,41 @@ impl WindowBuilder {
     }
 
     pub fn set_size(&mut self, size: Size) {
-        self.size = size;
+        // zero sized window results in server error
```

Comment:
Actually... zero width *or* zero height is not allowed. You only check for "*and*".